src/components/Article/Article.jsx

```diff
@@ -1,38 +1,79 @@
 import { useEffect, useState } from "react";
 import { useParams } from "react-router-dom";
-import { getArticleById } from "../../utils/util-api-calls";
+import { getArticleById, voteOnArticle } from "../../utils/util-api-calls";
 import { capitaliseTopic, formatDate } from "../../utils/util-data-formatters";
-import {
-	HiBookmarkSquare,
-	HiMiniUserCircle,
-	HiCalendarDays,
-	HiArrowUp,
-	HiArrowDown,
-} from "react-icons/hi2";
+import { HiBookmarkSquare, HiMiniUserCircle, HiCalendarDays, HiArrowUp, HiArrowDown } from "react-icons/hi2";
 import "./Article.css";
 import Comments from "../Comments/Comments";
 
 function Article() {
 	const { article_id } = useParams();
-	const [articleData, setArticle] = useState([]);
+	const [articleData, setArticleData] = useState([]);
 	const [articleLoading, setArticleLoading] = useState(true);
+	const [articleVote, setArticleVote] = useState(0);
 
 	useEffect(() => {
 		setArticleLoading(true);
 		getArticleById(article_id)
 			.then((res) => {
 				setArticleLoading(false);
-				setArticle(res.data.article);
+				setArticleData(res.data.article);
 			})
 			.catch((error) => {
 				console.log(error);
 			});
 	}, []);
 
+	function handleUpVote() {
+		const initialVote = articleVote;
+		if (articleVote === -1) {
+			setArticleVote(1);
+			voteOnArticle(article_id, 2).catch((error) => {
+				console.log(error);
+				setArticleVote(initialVote);
+			});
+		} else if (articleVote === 0) {
+			setArticleVote(1);
+			voteOnArticle(article_id, 1).catch((error) => {
+				console.log(error);
+				setArticleVote(initialVote);
+			});
+		} else if (articleVote === 1) {
+			setArticleVote(0);
+			voteOnArticle(article_id, -1).catch((error) => {
+				console.log(error);
+				setArticleVote(initialVote);
+			});
+		}
+	}
+
+	function handleDownVote() {
+		const initialVote = articleVote;
+		if (articleVote === 1) {
+			setArticleVote(-1);
+			voteOnArticle(article_id, -2).catch((error) => {
+				console.log(error);
+				setArticleVote(initialVote);
+			});
+		} else if (articleVote === 0) {
+			setArticleVote(-1);
+			voteOnArticle(article_id, -1).catch((error) => {
+				console.log(error);
+				setArticleVote(initialVote);
+			});
+		} else if (articleVote === -1) {
+			setArticleVote(0);
+			voteOnArticle(article_id, 1).catch((error) => {
+				console.log(error);
+				setArticleVote(initialVote);
+			});
+		}
+	}
```

Comment:
Here we have two quite similar functions with a lot of repeated code - would it be possible to refactor these into one? 🤔